kale/embed/video_se_i3d.py

```diff
@@ -0,0 +1,231 @@
+            model.Mixed_4f.add_module("SELayerTCc", SELayerC(832))
+            model.Mixed_5b.add_module("SELayerTCc", SELayerC(832))
+            model.Mixed_5c.add_module("SELayerTCc", SELayerC(1024))
+
+        self.model = model
+
+    def forward(self, x):
+        return self.model(x)
+
+
+class SEInceptionI3DFlow(nn.Module):
+    def __init__(self, num_channels, num_classes, attention):
+        super(SEInceptionI3DFlow, self).__init__()
+        model = InceptionI3d(in_channels=num_channels, num_classes=num_classes)
+        # n = 16
+        # if attention == "SELayerC":
+        #     model.Mixed_3b.add_module("SELayerC", SELayerC(256))
+        #     model.Mixed_3c.add_module("SELayerC", SELayerC(480))
+        #     model.Mixed_4b.add_module("SELayerC", SELayerC(512))
+        #     model.Mixed_4c.add_module("SELayerC", SELayerC(512))
+        #     model.Mixed_4d.add_module("SELayerC", SELayerC(512))
+        #     model.Mixed_4e.add_module("SELayerC", SELayerC(528))
+        #     model.Mixed_4f.add_module("SELayerC", SELayerC(832))
+        #     model.Mixed_5b.add_module("SELayerC", SELayerC(832))
+        #     model.Mixed_5c.add_module("SELayerC", SELayerC(1024))
+        #
+        # elif attention == "SELayerT":
+        #     model.Mixed_3b.add_module("SELayerT", SELayerT(n // 4))
+        #     model.Mixed_3c.add_module("SELayerT", SELayerT(n // 4))
+        #     model.Mixed_4b.add_module("SELayerT", SELayerT(n // 8))
+        #     model.Mixed_4c.add_module("SELayerT", SELayerT(n // 8))
+        #     model.Mixed_4d.add_module("SELayerT", SELayerT(n // 8))
+        #     model.Mixed_4e.add_module("SELayerT", SELayerT(n // 8))
+        #     model.Mixed_4f.add_module("SELayerT", SELayerT(n // 8))
+
+        self.model = model
+
+    def forward(self, x):
+        return self.model(x)
+
+
+def se_inception_i3d(name, num_channels, num_classes, attention, pretrained=False, progress=True, rgb=True):
+    """Get InceptionI3d module w/o pretrained model."""
+    if rgb:
+        model = SEInceptionI3DRGB(num_channels, num_classes, attention)
+    else:
+        model = SEInceptionI3DFlow(num_channels, num_classes, attention)
+
+    if pretrained:
+        state_dict = load_state_dict_from_url(model_urls[name], progress=progress)
+
+        # delete logits.conv3d parameters due to different class number.
+        # state_dict.pop("logits.conv3d.weight")
+        # state_dict.pop("logits.conv3d.bias")
+
+        # Create new OrderedDict that add `model.`
+        from collections import OrderedDict
+
+        new_state_dict = OrderedDict()
+        for k, v in state_dict.items():
+            name = "model.{}".format(k)
+            new_state_dict[name] = v
+
+        # Load params except SELayer
+        model.load_state_dict(new_state_dict, strict=False)
+    return model
+
+
+def se_i3d_joint(rgb_pt, flow_pt, num_classes, attention, pretrained=False, progress=True):
+    """Get I3D models."""
```

Comment:
too brief, what `se` and `joint` mean?

Reply:
"se" means SELayer and "joint" means different inputs like RGB, flow and joint.

Reply:
The updated docstrings is helpful.